scripts/o3de/o3de/repo_properties.py

```diff
@@ -92,16 +94,52 @@ def create_remote_object_archive(src_data_path: pathlib.Path,
         FileNotFoundError: If the json_data_path does not exist.
         ValueError: If the json_data_path is not a dict.
     """
-    if download_prefix is None:
-        logger.error('The --download-prefix argument must be provided. A url prefix for a file attached to a Github release might look like this:'
-                     '-dp https://github.com/o3de/o3de-extras/releases/download/2305.0')
-        return {}
     zip_path = releases_path / archive_filename
     # check if a object.zip folder already exist in the path - ask user if they want to overwrite the current zip
     if not force and zip_path.exists():
         logger.error(f'{zip_path} already exists.  Use --force command to overwrite the existing zip or '
                      'provide a new location to save your archive.')
         return {}
+    if not upload_git_release_tag and not download_prefix:
+        logger.error('You need to provide a --upload-github-release-tag if you want to upload to github.\n'
+                     'If you want to use other version control systems you must provide --download-prefix argument\n'
+                     'A url prefix for a file attached to a Github release might look like this:'
+                     '-dp https://github.com/o3de/o3de-extras/releases/download/2305.0')
+        return {}
+    if upload_git_release_tag and download_prefix:
+        logger.error('Upload to GitHub will automatically generate a download prefix for you.\n'
+                     'Use the `--download-prefix` CLI only when you want to use other version control system.')
+        return {}
+    if upload_git_release_tag and not download_prefix:
+        repo_uri = urllib.parse.urlparse(repo_uri)
+        git_provider = github_utils.get_github_provider(repo_uri)
+        if git_provider:
+        # Parse the URL to extract owner and repository name
+            owner_repo = repo_uri.geturl().split("github.com/")[1].rstrip('.git').split('/')
+            owner = owner_repo[0]
+            repo = owner_repo[1]
+        else:
+            logger.error(f'Failed to determine Git provider from {repo_uri}.')
+            return {}
+        download_prefix = f'https://github.com/{owner}/{repo}/releases/download/{upload_git_release_tag}/{archive_filename}'
+     
+        json_data = merge_json_data(json_data_path, {
+            'repo_uri': repo_uri,
+            'download_source_uri': download_prefix
+        })
+
+        logging.info(f"Creating '{download_prefix}'")
+
+        shutil.make_archive(releases_path / pathlib.Path(archive_filename).stem, 'zip', src_data_path)
+        with zip_path.open('rb') as f:
+            json_data['sha256'] = hashlib.sha256(f.read()).hexdigest()
+        
+        # Upload release archive zip to Github
+        if git_provider.upload_release_to_github(repo_uri, zip_path, archive_filename, upload_git_release_tag) == 1:
```

Comment:
After changing the function name you should be able to call it like this
```suggestion
        if git_provider.upload_release(repo_uri, zip_path, archive_filename, upload_git_release_tag) == 1:
```